airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,546 @@
+        with open(self.kubernetes_jwt_path) as f:
+            jwt = f.read()
+            _client.auth_kubernetes(role=self.kubernetes_role, jwt=jwt)
+
+    def _auth_github(self, _client: hvac.Client) -> None:
+        _client.auth.github.login(token=self.token)
+
+    def _auth_gcp(self, _client: hvac.Client) -> None:
+        # noinspection PyProtectedMember
+        from airflow.providers.google.cloud.utils.credentials_provider import (
+            get_credentials_and_project_id,
+            _get_scopes
+        )
+        scopes = _get_scopes(self.gcp_scopes)
+        credentials, _ = get_credentials_and_project_id(key_path=self.gcp_key_path,
+                                                        scopes=scopes)
+        _client.auth.gcp.configure(credentials=credentials)
+
+    def _auth_approle(self, _client: hvac.Client) -> None:
+        _client.auth_approle(role_id=self.role_id, secret_id=self.secret_id)
+
+    def get_secret(self, secret_path: str, secret_version: Optional[int] = None) -> Optional[dict]:
+        """
+        Get secret value from the engine.
+
+        :param secret_path: The path of the secret.
+        :type secret_path: str
+        :param secret_version: Specifies the version of Secret to return. If not set, the latest
+            version is returned. (Can only be used in case of version 2 of KV).
+        :type secret_version: int
+
+        See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v1.html
+        and https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
+
+        :return: secret stored in the vault as a dictionary
+        """
+        try:
+            if self.kv_engine_version == 1:
+                if secret_version:
+                    raise VaultError("Secret version can only be used with version 2 of the KV engine")
+                response = self.client.secrets.kv.v1.read_secret(
+                    path=secret_path, mount_point=self.mount_point)
+            else:
+                response = self.client.secrets.kv.v2.read_secret_version(
+                    path=secret_path, mount_point=self.mount_point, version=secret_version)
+        except InvalidPath:
+            self.log.debug("Secret not found %s with mount point %s", secret_path, self.mount_point)
+            return None
+
+        return_data = response["data"] if self.kv_engine_version == 1 else response["data"]["data"]
+        return return_data
+
+    def get_secret_metadata(self, secret_path: str) -> Optional[dict]:
+        """
+        Reads secret metadata (including versions) from the engine. It is only valid for KV version 2.
+
+        :param secret_path: The path of the secret.
+        :type secret_path: str
+        :rtype: dict
+        :return: secret metadata. This is a Dict containing metadata for the secret.
+
+                 See https://hvac.readthedocs.io/en/stable/usage/secrets_engines/kv_v2.html for details.
```

Comment:
Do we need this "tab space", if 'yes' then we should add it to L257 too

Reply:
Nope